store/etcd/etcd.go

```diff
@@ -160,6 +160,9 @@ func (s *Etcd) Put(key string, value []byte, opts *store.WriteOptions) error {
 // Delete a value at "key"
 func (s *Etcd) Delete(key string) error {
 	_, err := s.client.Delete(store.Normalize(key), false)
+	if isKeyNotFoundError(err) {
+		return nil
```

Comment:
I'd say at least for Consul and leave the tests as it is so that we don't enforce the same error on `boltdb`. We can do `boltdb` in another PR to fix it and add the tests on `store.ErrKeyNotFound` :)